### PR TITLE
example in README and documentation

### PR DESCRIPTION
**Pull request by phip1611**
I experienced that this library is not completely trivial to use at a first glance. For example I was confused with the file-abstraction and how to use it. I'd highly appreciate it if a small code example can be included in both, README and Rustdoc. This will help others to quickly understand the design and usage of this crate.

What do you think?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,25 @@ For reading files, it provides multiple levels of support:
...
 Supported file formats: ELF, Mach-O, Windows PE/COFF, Wasm, and Unix archive.
 
+## Example
+```rust
+use std::fs::File;
+use std::io::Read;
+use object::{Object, ObjectSection, File as ObjectFile};
+
+/// Reads an ELF-file and displays the content of the ".boot" section.
+fn main() {
```
Comment (philipc) [inline README.md:23]:
```suggestion
fn main() -> Result<(), Box<dyn Error>> {
```
and replace all of the `unwrap` with `?`.

File: README.md [hunk 2]
```diff
@@ -13,6 +13,25 @@ For reading files, it provides multiple levels of support:
...
+    let mut file = File::open("./multiboot2-binary.elf").unwrap();
+    let mut data = vec![];
+    file.read_to_end(&mut data).unwrap();
+    let data = data.into_boxed_slice();
```
Comment (philipc) [inline README.md:27]:
```suggestion
    let data = fs::read("<path-to-file>")?
```

File: README.md [hunk 3]
```diff
@@ -13,6 +13,25 @@ For reading files, it provides multiple levels of support:
...
+    let obj_file = ObjectFile::parse(&*data).unwrap();
```
Comment (philipc) [inline README.md:28]:
```suggestion
    let obj_file = object::File::parse(&*data).unwrap();
```

File: src/lib.rs [hunk 4]
```diff
@@ -25,6 +25,25 @@
 //! ## Unified write API
 //!
 //! [write::Object] allows building an object and then writing it out.
+//!
+//! ## Example
```
Comment (philipc) [inline src/lib.rs:29]:
```suggestion
//! ## Example for unified read API
```

**Review by philipc — CHANGES_REQUESTED** [review]
Thanks! I agree this will be useful.

Let's leave out `examples/read_section_from_elf.rs` though. It isn't particularly useful to be able to run this with a fixed filename, and `objdump.rs` already serves as a more extensive example.

File: README.md [hunk 5]
```diff
@@ -13,6 +13,25 @@ For reading files, it provides multiple levels of support:
 
 Supported file formats: ELF, Mach-O, Windows PE/COFF, Wasm, and Unix archive.
 
+## Example
+```rust
+use std::fs::File;
+use std::io::Read;
+use object::{Object, ObjectSection, File as ObjectFile};
+
+/// Reads an ELF-file and displays the content of the ".boot" section.
```
Comment (philipc) [inline README.md:22]:
```suggestion
/// Reads a file and displays the content of the ".boot" section.
```

File: README.md (hunk 2 above)
Comment (phip1611) [reply README.md]: Oh great! After almost 3 years of doing Rust, I didn't know about `fs::read` and always did the other, more laborious way - thanks!

**Comment by phip1611** [comment]
Hi @philipc
I have tried to fulfill all your suggestions for improvement. What do you think?

File: src/lib.rs [hunk 6]
```diff
@@ -25,6 +25,25 @@
 //! ## Unified write API
 //!
 //! [write::Object] allows building an object and then writing it out.
+//!
+//! ## Example for unified read API
+//!  ```rust
```
Comment (philipc) [inline src/lib.rs:30]:
```suggestion
//!  ```no_run
```

**Comment by philipc** [comment]
Looks good once the tests are passing.

**Review by philipc — APPROVED** [review]

**Pull request merged**